flask_saml_sso/sso.py

```diff
@@ -6,128 +6,33 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
 import contextlib
+import functools
+import logging
 import os
 from urllib import parse
 
 import flask
-import functools
 from onelogin.saml2.auth import OneLogin_Saml2_Auth
 from onelogin.saml2.constants import OneLogin_Saml2_Constants
-from onelogin.saml2.idp_metadata_parser import OneLogin_Saml2_IdPMetadataParser
 from onelogin.saml2.response import OneLogin_Saml2_Response
+from onelogin.saml2.settings import OneLogin_Saml2_Settings
 from onelogin.saml2.xml_utils import OneLogin_Saml2_XML
 from werkzeug import exceptions
 
 from . import session
+from . import settings
+
+logger = logging.getLogger('flask_saml_sso')
```

Comment:
If we hardcode the logger name, we cannot distinguish which file a message was logged from. The pattern I know of is `logger = logging.getLogger(__name__)`. The name is hardcoded in a few files.

Reply:
I hardcoded this on purpose, to make every log message originate from the auth library as a whole, as users of the library probably aren't interesting in the internals of it, and of which log messages are from which files.

Reply:
Huh. That sounds like a good idea. Although it would require a second `debug_logger` object, because in that case you definitely want `__name__`. I would be interested to know  if you have seen this elsewhere :-)